src/edu/washington/escience/myria/operator/SampledDbInsertTemp.java

```diff
@@ -0,0 +1,221 @@
+import com.almworks.sqlite4java.SQLiteException;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.*;
+import edu.washington.escience.myria.accessmethod.AccessMethod;
+import edu.washington.escience.myria.accessmethod.ConnectionInfo;
+import edu.washington.escience.myria.accessmethod.SQLiteInfo;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.IntColumnBuilder;
+import edu.washington.escience.myria.parallel.RelationWriteMetadata;
+import edu.washington.escience.myria.storage.MutableTupleBuffer;
+import edu.washington.escience.myria.storage.TupleBatch;
+
+/**
+ * Samples the stream into a temp relation.
+ */
+public class SampledDbInsertTemp extends UnaryOperator implements DbWriter {
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /** The connection to the database database. */
+  private AccessMethod accessMethod;
+  /** The information for the database connection. */
+  private ConnectionInfo connectionInfo;
+  /** The name of the table the tuples should be inserted into. */
+  private final RelationKey sampleRelationKey;
+  /** The name of the table the tuples should be inserted into. */
+  private final RelationKey countRelationKey;
+
+  /** Total number of tuples seen from the child. */
+  private int tupleCount = 0;
+  /** Number of tuples to sample from the stream. */
+  private final int streamSampleSize;
+  /** Reservoir that holds sampleSize number of tuples. */
+  private MutableTupleBuffer reservoir = null;
+  /** Sampled tuples ready to be returned. */
+  private List<TupleBatch> batches;
+  /** Next element of batches List that will be returned. */
+  private int batchNum = 0;
+  /** True if all samples have been gathered from the child. */
+  private boolean doneSamplingFromChild;
+
+  /** The output schema. */
+  private static final Schema COUNT_SCHEMA = Schema.of(
+      ImmutableList.of(Type.INT_TYPE, Type.INT_TYPE, Type.INT_TYPE),
+      ImmutableList.of("WorkerID", "PartitionSize", "StreamSize"));
+
+  /**
+   * @param child
+   *          the source of tuples to be inserted.
+   * @param streamSampleSize
+   *          number of tuples to store from the stream
+   * @param sampleRelationKey
+   *          the key of the table that the tuples should be inserted into.
+   * @param countRelationKey
+   *          the key of the table that the tuple counts will be inserted into.
+   * @param connectionInfo
+   *          the parameters of the database connection.
+   */
+  public SampledDbInsertTemp(final Operator child, final int streamSampleSize,
+      final RelationKey sampleRelationKey, final RelationKey countRelationKey,
+      final ConnectionInfo connectionInfo) {
+    super(child);
+    // Sampling setup.
+    Preconditions.checkArgument(streamSampleSize >= 0L,
```

Comment:
Nitpick: `L` qualifier is unnecessary for comparing `int` variable to `int` literal.